clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +34,80 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	defer cleanupResources(ctx)
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		if err := ste.ExecuteStep(ctx, &conf.Steps[i]); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
-	return nil
+	var err error
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		var lock sync.Mutex
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				if phaseErr := ste.ExecutePhase(ctx, phase); phaseErr != nil {
+					lock.Lock()
+					defer lock.Unlock()
+					if err == nil {
+						err = phaseErr
+					}
+				}
+			})
+		}
+		wg.Wait()
+	}
+	return err
 }
 
 // ExecutePhase executes single test phase based on provided phase configuration.
 func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) error {
+	if phase.NamespaceRange == nil {
+		for j := range phase.ObjectBundle {
+			ste.ExecuteObject(ctx, "", phase.ReplicasPerNamespace, &phase.ObjectBundle[j])
+		}
+	} else {
+		nsBasename := framework.AutomanagedNamespaceName
+		if phase.NamespaceRange.Basename != nil {
+			nsBasename = *phase.NamespaceRange.Basename
+		}
+		for i := phase.NamespaceRange.Min; i <= phase.NamespaceRange.Max; i++ {
+			nsName := fmt.Sprintf("%v-%d", nsBasename, i)
+			for j := range phase.ObjectBundle {
+				if err := ste.ExecuteObject(ctx, nsName, phase.ReplicasPerNamespace, &phase.ObjectBundle[j]); err != nil {
+					return err
```

Comment:
Let's not fail fast (like we discussed earlier why we should try to avoid that in our scalability tests). Just record the error and skip creating any further objects in this particular bundle (we can still continue for other namespaces).

In fact, you can even continue at a lower granularity (i.e for other replicas of the objectBundle within the same namespace). And for that you'll need to change the `ExecuteObject` function to take a single replicaIndex and create only that one instance (instead of creating `replicaCount` many instances). Wdyt?

Reply:
I have two assumptions:
* All errors will be logged.
* Only critical error will be returned from the function. After critical error, test should be terminated.

So, in case `ExecuteObject` fails, it will log error and return `nil`. This way test will not fail fast unless it is needed.

Reply:
On the other hand I can imagine other implementation that returns `[]error` from each `Execute...` function.
TBH, I'm not sure which one is better - both have pros and cons.

Reply:
What's your criteria to decide if something is a critical error?

Reply:
At this point I'm not sure what should be the criteria.
There is a function that decides whether given err list is critical (with TODO status).